### PR TITLE
WIP implement intercept fitting in celer

### PR DESCRIPTION
**Pull request by mathurinm**
For the Lasso, it is OK to center and scale X, and ceter y, then to revert the transformation for coefs and intercept. In the sparse case this is tricky without breaking the sparsity pattern of X, but celer_path solver supports it.

### CODE REVIEW

**Comment by jolars** [comment]
Centering y is fine when fitting the intercept, I agree, but not when not fitting the intercept, right?

Also, we cannot scale X, since it affects the coefficients, right? We would have to do that for all of the solvers. glmnet also does just-in-time standardization for the sparse case, by the way, but it standardizes with the population standard deviation.

File: solvers/celer.py [hunk 1]
```diff
@@ -20,19 +22,28 @@ class Solver(BaseSolver):
...
-    def skip(self, X, y, lambdas, fit_intercept):
-        if fit_intercept:
-            return True, f"{self.name} does not handle fit_intercept"
-
-        return False, None
-
     def run(self, tol):
         warnings.filterwarnings("ignore", category=ConvergenceWarning)
 
+        if self.fit_intercept:
```
Comment (jolars) [inline solvers/celer.py:45]: I guess there should be something in this block?

**Comment by mathurinm** [comment]
Yes you're right for normalization, this is a different issue. 
Indeed centering y should only happen for fit_intercept, and this can be done at the solver level (other solvers may implement intercept fitting directly)
this is a WIP, I will get back to it later

**Comment by jolars** [comment]
Sounds good! Thanks for the PR!

**Comment by mathurinm** [comment]
Take over in #2

**Pull request closed without merging**